### ISSUE TITLE
Wrong display in settings page

### ISSUE THREAD

**Issue by nikiie8** [wallabag/wallabag#4294]
### Issue details

Hi all
My settings page looks very strange. https://i.postimg.cc/fLkskWnT/Screen.png
I can not change anything. Other pages looks as usual. Wallabag works perfectly as well.

### Environment
I have installed wallabag 2.3.8 in my VPS via download package. Initially it doesn't work  due to issue #4271  as I have PHP 7.4. But I fixed it as it was written in [wallabag/php-mobi@bfe9c18](https://github.com/wallabag/php-mobi/commit/bfe9c18d038e5ca24664cfdab107a645ac57c9b5).
OS: Debian 9
I have used MariaBD database
I have tried to open wallabag in Firefox, Chrome and IE. Everything is the same.

**Comment by Paulikid** [comment]
same here after updating wallabag the last time. I also fixed the above issue with PHP 7.4 editing "MOBIClass/RecordFactory.php" and"composer.json". 

Is there anything we can do to fix this?

**Comment by j0k3r** [comment]
Could you check the console / networks tab of your browser to see if some css aren't loaded properly?

**Comment by flow7** [comment]
Exact same thing here -- @j0k3r as far as I can tell material.css was properly loaded.
Anything specific you need to see on this?

**Comment by j0k3r** [comment]
Anything that can be a 404 in the console network

**Comment by flow7** [comment]
No 404's as far as I can see ;)

**Comment by Paulikid** [comment]
Same here, material. css loaded without any errors, no 404s noticeable.

**Comment by leolivier** [comment]
Not exactly the same for me.
I also applied the php7.4 patch above and reinstalled 2.3.8. 
Now, the settings page is also weirdly displayed but also the display of articles in grid mode is changed so that I have only one article per line instead of 3 (or 4?) usually.  
Tried in Firefox 75 and Edge 44. 
Everything is loaded with a 200 code.

**Comment by leolivier** [comment]
Hi
Not exactly the same error for me:
I also applied the php7.4 patch above and reinstalled 2.3.8. 
Now, the settings page is also weirdly displayed but also the display of articles in grid mode is changed so that I have only one article per line instead of 3 (or 4?) usually.  
Tried in Firefox 75 and Edge 44. 
Everything is loaded with a 200 code.

**Comment by TalmarGrosskotz** [comment]
Why is this closed? Is there any solution available that I am not aware off?

**Comment by leolivier** [comment]
> 
> 
> Why is this closed? Is there any solution available that I am not aware off?

This is not closed, your's references this one which is still open...

By the way, @j0k3r,  its working again for me after I migrated to the docker version, god only knows why...

**Comment by quent1-fr** [comment]
Found the bug. The problem was caused by Twig 1.x not being compatible with PHP 7.4 and removing some whitespaces (see https://github.com/twigphp/Twig/issues/3248 for more details).

Upgrading twig/twig dependency to version 1.42 did the trick.

**Comment by anothervictim** [comment]
> Found the bug. The problem was caused by Twig 1.x not being compatible with PHP 7.4 and removing some whitespaces (see [twigphp/Twig#3248](https://github.com/twigphp/Twig/issues/3248) for more details).
> 
> Upgrading twig/twig dependency to version 1.42 did the trick.

Works for me. Much thanks!

**Comment by Paulikid** [comment]
> Found the bug. The problem was caused by Twig 1.x not being compatible with PHP 7.4 and removing some whitespaces (see [twigphp/Twig#3248](https://github.com/twigphp/Twig/issues/3248) for more details).
> 
> Upgrading twig/twig dependency to version 1.42 did the trick.

Sorry if this might be a dumb question, but how do I upgrade twig/twig dependency to version 1.42?

**Comment by j0k3r** [comment]
We’ll soon push a PR to fix that in the master branch.

**Comment by quent1-fr** [comment]
> Sorry if this might be a dumb question, but how do I upgrade twig/twig dependency to version 1.42?

Tried to update through composer with no luck (composer got killed by my system after using ~2GB of RAM...). I did it the ugly way by downloading twig/twig:1.4 on a new folder then replacing it on wallabag.

To do so, in a shell, type : 

```
mkdir /tmp/twig1.4
cd /tmp/twig1.4
composer require twig/twig:1.4
```

Once finished, replace the `vendor/twig/twig` folder on wallabag with the one from `/tmp/twig1.4/vendor/twig/twig`.

You can then remove `/tmp/twig1.4`.

**Comment by j0k3r** [comment]
Of course the Twig version installed on wallabag 2.3.8 is quite old.
Also, the 2.3.8 isn't compatible with PHP 7.4.

The upcoming 2.4 will fix that issue because it use Twig 2.

**Comment by Paulikid** [comment]
> composer require twig/twig:1.4

after doing this my wallabag doesn't load anymore, I always get a "HTTP Error 500"

**Comment by Paulikid** [comment]
> Of course the Twig version installed on wallabag 2.3.8 is quite old.
> Also, the 2.3.8 isn't compatible with PHP 7.4.
> 
> The upcoming 2.4 will fix that issue because it use Twig 2.
> 
> In the meantime, to fix the issue, you can run `SYMFONY_ENV=prod composer update twig/twig --no-dev --prefer-fist`.
> Let me know if it's work.

Thank you very much, this worked like a charm. 

btw the command contains a typo, the option must be "--prefer-dist"

**Comment by leolivier** [comment]
> btw the command contains a typo, the option must be "--prefer-dist"

"--prefer-fist" is way funnier 🤣 🤣 🤣

**Comment by nikiie8** [comment]
Thanks. Now works like a charm!

**Comment by j0k3r** [comment]
As we have a fix and that issue is fixed in 2.4.0, I'm closing that issue.
